test/core/runtime/block_builder_test.cpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "runtime/impl/block_builder_impl.hpp"
+
+#include <gtest/gtest.h>
+#include "core/runtime/runtime_test.hpp"
+#include "core/storage/merkle/mock_trie_db.hpp"
+#include "extensions/extension_impl.hpp"
+#include "primitives/impl/scale_codec_impl.hpp"
+#include "runtime/impl/wasm_memory_impl.hpp"
+#include "testutil/outcome.hpp"
+#include "testutil/runtime/wasm_test.hpp"
+
+using namespace testing;
+using kagome::common::Buffer;
+using kagome::extensions::ExtensionImpl;
+using kagome::primitives::Extrinsic;
+using kagome::primitives::InherentData;
+using kagome::primitives::ScaleCodecImpl;
+using kagome::runtime::BlockBuilder;
+using kagome::runtime::BlockBuilderImpl;
+using kagome::runtime::WasmMemoryImpl;
+using kagome::storage::merkle::MockTrieDb;
+
+class BlockBuilderTest : public RuntimeTest {
+ public:
+  void SetUp() override {
+    RuntimeTest::SetUp();
+
+    builder_ =
+        std::make_unique<BlockBuilderImpl>(state_code_, extension_, codec_);
+  }
+
+ protected:
+  std::unique_ptr<BlockBuilder> builder_;
+};
+
+/**
+ * @given block builder
+ * @when calling check_inherents runtime function
+ * @then the result of the check is obtained given that the provided arguments were valid
```

Comment:
Please, elaborate more: what arguments are used in that case in in the one below, and what exactly is expected

Reply:
Tests for Runtime API are still incomplete, so I think when the actual arguments that should be passed to WASM are known, it would make sense to make tests more detailed.